src/utils/source.js

```diff
@@ -293,9 +293,14 @@ function getMode(source: Source, sourceMetaData: SourceMetaDataType) {
   return { name: "text" };
 }
 
-function isLoaded(source: SourceRecord) {
+function isLoaded(source: ?SourceRecord) {
+  if (!source) {
```

Comment:
should this ever happen?

Reply:
it could if the caller doesn't check if there's a source... Dropped it because the places we use isLoaded should be safe now.